src/Components/Medicine/MedicineAdministration.tsx

```diff
@@ -47,13 +56,34 @@ export default function MedicineAdministration(props: Props) {
   }, [props.prescriptions]);
 
   const handleSubmit = async () => {
-    const administrations = prescriptions
-      .map((prescription, i) => ({
-        prescription,
-        notes: notes[i],
-        administered_date: isCustomTime[i] ? customTime[i] : undefined,
-      }))
-      .filter((_, i) => shouldAdminister[i]);
+    const administrations = [];
+
+    for (let i = 0; i < prescriptions.length; i++) {
+      if (shouldAdminister[i]) {
+        if (prescriptions[i].dosage_type === "TITRATED") {
+          const errorText = AdministrationDosageValidator(
+            dosages[i].dosage,
+            prescriptions[i].base_dosage,
+            prescriptions[i].target_dosage
+          );
+          if (errorText) {
+            setDosages((dosages) => {
+              const newDosages = [...dosages];
+              newDosages[i].error = errorText;
+              return newDosages;
+            });
+            return;
+          }
```

Comment:
The error is not being removed/set to undefined once it's resolved.

```suggestion
          const error = AdministrationDosageValidator(
            prescriptions[i].base_dosage,
            prescriptions[i].target_dosage
          )(dosages[i].dosage)
          setDosages((dosages) => {
            const newDosages = [...dosages];
            newDosages[i].error = error;
            return newDosages;
          });
          if (error) {
            return;
          }
```